crates/oxc_mangler/src/keep_names.rs

```diff
@@ -3,24 +3,75 @@ use oxc_ast::{AstKind, ast::*};
 use oxc_semantic::{AstNode, AstNodes, ReferenceId, Scoping, SymbolId};
 use rustc_hash::FxHashSet;
 
-#[cfg_attr(not(test), expect(dead_code))]
-pub fn collect_name_symbols(scoping: &Scoping, ast_nodes: &AstNodes) -> FxHashSet<SymbolId> {
-    let collector = NameSymbolCollector::new(scoping, ast_nodes);
+#[derive(Debug, Clone, Copy, Default)]
+pub struct MangleOptionsKeepNames {
+    /// Keep function names so that `Function.prototype.name` is preserved.
```

Comment:
I don't think this is comment is . It's not a property of `Function.prototype`, but a property of the `Function` instance. "`name` property" might be a better description.

Ditto `Class.prototype.name` comment below.

Sorry to be a pedant!